File: src/main/java/de/tum/in/www1/artemis/web/rest/dto/CoursesForDashboardDTO.java
package de.tum.in.www1.artemis.web.rest.dto;

import java.util.Set;

import com.fasterxml.jackson.annotation.JsonInclude;

import de.tum.in.www1.artemis.domain.exam.Exam;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public record CoursesForDashboardDTO(Set<CourseForDashboardDTO> courses, Set<Exam> activeExams) {

}

File: src/main/webapp/app/core/course/shared/entities/courses-for-dashboard-dto.ts
import { CourseForDashboardDTO } from 'app/core/course/shared/entities/course-for-dashboard-dto';
import { Exam } from 'app/exam/shared/entities/exam.model';

export class CoursesForDashboardDTO {
    courses: CourseForDashboardDTO[];
    activeExams: Exam[];
}
